__tests__/app.test.js

```diff
@@ -70,3 +70,57 @@ describe("GET /api", () => {
       });
   });
 });
+
+describe("GET /api/articles/:articleid", () => {
+  test("responds with status code 200", () => {
+    return request(app).get("/api/articles/6").expect(200);
+  });
+  test("responds with an article object", () => {
+    return request(app)
+      .get("/api/articles/5")
+      .then(({ body }) => {
+        expect(typeof body).toBe("object");
+      });
+  });
+  test("responds with an article object with the correct properties", () => {
+    return request(app)
+      .get("/api/articles/2")
+      .then(({ body }) => {
+        expect(body.article.hasOwnProperty("title")).toBe(true);
+        expect(body.article.hasOwnProperty("article_id")).toBe(true);
+        expect(body.article.hasOwnProperty("body")).toBe(true);
+        expect(body.article.hasOwnProperty("topic")).toBe(true);
+        expect(body.article.hasOwnProperty("created_at")).toBe(true);
+        expect(body.article.hasOwnProperty("votes")).toBe(true);
+        expect(body.article.hasOwnProperty("article_img_url")).toBe(true);
+      });
+  });
+  test("responds with the correct article object", () => {
+    const expected = {
+      article_id: 7,
+      article_img_url:
+        "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+      author: "icellusedkars",
+      body: "I was hungry.",
+      created_at: "2020-01-07T14:08:00.000Z",
+      title: "Z",
+      topic: "mitch",
+      votes: 0,
+    };
+    return request(app)
+      .get("/api/articles/7")
+      .then(({ body }) => {
+        expect(body.article).toEqual(expected);
+      });
+  });
+  test("responds with a 404 code when passed an invalid article id", () => {
+    return request(app).get("/api/articles/50").expect(404);
+  });
+  test("responds with appropriate error message when passed invalid article id", () => {
+    return request(app)
+      .get("/api/articles/50")
+      .then(({ text }) => {
+        expect(text).toBe("No user found for user_id: 50");
+      });
```

Comment:
I'd respond on the key of message or something similar, but text isn't a terrible choice as long as its consistent

Reply:
Also this message is completely inappropriate for the error that's happening